### PR TITLE
Fixes and enhancements to rolling fama-french regression

### PR DESCRIPTION
**Pull request by twiecki**
`rolling_fama_french` is actually more general, as one can also pass in their own risk-factors. I thus renamed the function (by default it will still load FF). While doing that I also found a couple of other problems that I fixed here.

### CODE REVIEW

File: pyfolio/timeseries.py [hunk 1]
```diff
@@ -547,7 +547,7 @@ def rolling_beta(returns, factor_returns,
         return out
 
 
-def rolling_fama_french(returns, factor_returns=None,
+def rolling_correlation(returns, factor_returns=None,
                         rolling_window=APPROX_BDAYS_PER_MONTH * 6):
     """
     Computes rolling Fama-French single factor betas using a multivariate
```
Comment (gusgordon) [inline pyfolio/timeseries.py]: This doc line should be updated too.

**Comment by gusgordon** [comment]
Looks good - I'm not super familiar with how the performance attribution works, but this also computes an alpha. Is there any overlapping functionality between perf_attrib and here that we could combine?

File: pyfolio/plotting.py [hunk 2]
```diff
@@ -184,7 +184,7 @@ def plot_rolling_fama_french(returns,
 
     ax.set_ylabel('Beta')
 
-    rolling_beta = timeseries.rolling_fama_french(
+    rolling_beta = timeseries.rolling_regression(
```
Comment (gusgordon) [inline pyfolio/plotting.py]: This should also be `rolling_correlation` not `rolling_regression` - missed that.

Comment (twiecki) [reply pyfolio/plotting.py]: Good catch, thanks.

Comment (twiecki) [reply pyfolio/plotting.py]: Actually should be `rolling_regression` throughout.

**Comment by twiecki** [comment]
OK, names should match up now.

**Comment by gusgordon** [comment]
@twiecki Ok, good to merge from my point of view, if you think this isn't an issue: https://github.com/quantopian/pyfolio/pull/444#discussion_r141683801

**Pull request merged**